Training attempt

JohnDNewman opened the pull request:
Added support to create datasets to train a gemini model on. 
Added support to tune a gemini model
Added support to use tuned model in alt text generation

JohnDNewman committed:
Create train_flan.py

first attempt at basic code to train the model.

JohnDNewman committed:
Training work

More work on training file, changed prompt a bit. No real results

JohnDNewman committed:
Added support to collect training data

JohnDNewman committed:
Updated files for training.

create_caption.py: changed gemini generate to call with training args
gemini.py: added arguments for training, added code to collect data if training
train_flan.py: Changed code to just be for cleaning up datasets and providing good data to them. Added a tkinter image and entry for data collection.

JohnDNewman committed:
Move dataset creation support to DataProcessor class

data_proccesor.py: added imports for functions needed for saving training dataset, added argument to determine whether data should be used in training, added branch for training in informative alt text and link alt text generation, added function _save_to_dataset to save url, prompt to dataset

JohnDNewman committed:
More stuff about training datasets

db,.csv, and .jsonl are datafiles.
train_flan renamed to training
training.py: Convert loose methods to a class, add exception checking for loading images and json data, add more stuff to tkinter window to make the annotation easier

JohnDNewman committed:
Added support to create fine tuned models

data_processor.py: added import of the trainer, removed uneeded imports, changed training flag behavior to make a trainer if training is true, else none, changed prompt collection to use Trainer class instead of having dedicated function, remove save_to_dataset as Trainer handles that operation
jsonl files: created during testing
training.py: changed add_to_dataset args to remove datasetfilename and add image type, filename is now made from modelname and image type, create a functional create_gemini_model function which takes in a dataset filename, loads it, converts to google TuningDataset, creates a tuning job on the client with that dataset and print the job, create check_tuning_job_status to check how a job is going, currently only checks a hardcoded job, but could not do that with 1 line change. Default behavior of training.py now lists models and checks the status of a training job.

JohnDNewman committed:
fine tune and use a model support

data_processor.py: hardcoded the usage of a tuned model. needs to be refined and updated to the new api as well
site_processor.py: I scared the module loader so now genai refers to its own thing so chagnge alias to genal
training.py: updated create gemini model to actually use the data instead of the strings"text_input", "output" repeatedly. Added support in check_tuing_job_status to check any job. changed default behavior to do a lot of things for testing.

JohnDNewman committed:
Added full support for using tuned model/s

data_processor.py: added arg to use tuned model to constructor, added attribute to determine usage of tuned model/s, moved prompts to prompt var instead of copying for a third time,added support to use the 'icran' tuned model
Deleted unused or unusable jsonl files
training.py: specified the imports to cut down on import files, added comments, added self to complte_dataset, Changed default behavior to be listing all tuned models.

aidenap21 commented on app/app_code/data_processor.py:
```diff
@@ -218,6 +247,20 @@ def _generate_link_description(self):
+            try:
+                if self._tuned:
+                    response = self._dummy_client.models.generate_content(
+                        model="tunedModels/test-tuned-model-icran2mmdiyb",#TODO: change tuned model to be the one trained for links
```
Does this cause issues right now since it isn't tuned for links?

aidenap21 reviewed (commented):
Just one question about a TODO comment left, otherwise seems good

JohnDNewman committed:
Change code from comment

training.py: remove the branch of using the tuned model to generate alt text for link images as the model is not tuned for that.
